kernel/src/engine/default/mod.rs

```diff
@@ -158,6 +158,16 @@ impl<E: TaskExecutor> Engine for DefaultEngine<E> {
     }
 }
 
+trait UrlExt {
+    fn is_presigned(&self) -> bool;
+}
+
+impl UrlExt for Url {
+    fn is_presigned(&self) -> bool {
+        matches!(self.scheme(), "http" | "https") && self.query().is_some()
```

Comment:
nit: docs

Reply:
so only material change is checking that the query is Some? maybe link docs to the guarantee?

Reply:
Updated the checks to explicitly consider some well known cases. Initially i thought  covering all cases would not be feasible, so just make the heuristic a bit more concrete. Then again, hopefully people will leet us know in case there are servers out there using different signing schemes.